source/vibe/inet/webform.d

```diff
@@ -259,7 +259,20 @@ private bool parseMultipartFormPart(InputStream stream, ref FormFields form, ref
 		auto data = cast(string)stream.readUntil(boundary);
 		form.addField(name, data);
 	}
-	return stream.readLine(max_line_length) != "--";
+	
+	ubyte[2] ub;
+	stream.read(ub.ptr[0 .. 2]);
+	if (ub == "--")
+	{
+		static NullOutputStream null_stream;
+		if (!null_stream)
+			null_stream = new NullOutputStream;
+
+		null_stream.write(stream);
+		return false;
+	}
+	stream.readLine(max_line_length);
```

Comment:
I think this should rather be `enforce(ub == cast(ubyte[])"\r\n");`. Any other text between the boundary and the newline is invalid AFAICS. Also, in this case it would skip the *next* line instead of the current, because the "\r\n" has already been read by `stream.read(ub)`.